### PR TITLE
[Backport 2.x] Always show security screen and shows error page when trying to access forbidden data-source (#1964)

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Manually backports #1964 

Manual backport is required because:
- Service Accounts is not present in 2.x line and there were changes made to service account related files on the commit that went into main.
- User page has conflict since some display text changed in the original commit.

### Check List
- [x] New functionality includes testing
~- [ ] New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by RyanL1997 — APPROVED** [review]

File: public/apps/configuration/panels/service-account-list.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
+/*
```
Comment (derek-ho) [inline public/apps/configuration/panels/service-account-list.tsx:1]: This file should not be here

File: public/apps/configuration/panels/test/__snapshots__/service-account-list.test.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (derek-ho) [inline public/apps/configuration/panels/test/__snapshots__/service-account-list.test.tsx.snap:1]: also pulled in

File: public/apps/configuration/panels/test/service-account-list.test.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
+/*
```
Comment (derek-ho) [inline public/apps/configuration/panels/test/service-account-list.test.tsx:1]: Also pulled in

**Comment by derek-ho** [comment]
We should delete service accounts on main if we have no plans to move forward with it, to prevent these pesky backport failures

**Comment by DarshitChanpura** [comment]
Removed.

**Review by derek-ho — APPROVED** [review]

**Pull request merged**